modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -131,6 +133,49 @@ public Settings getAdditionalIndexSettings(
                     return builder.build();
                 }
             }
+            if (templateIndexMode == IndexMode.LOGSDB
+                || IndexSettings.MODE.get(indexTemplateAndCreateRequestSettings) == IndexMode.LOGSDB) {
```

Comment:
I think this doesn't cover the case where logsdb is enabled by default for `log-*-*` pattern? This is kind of annoying. Maybe for now this isn't a very big deal. We would then require that both index mode and `index.logsdb.route_on_sort_fields` need to be configured.

Reply:
Correct.. moved the logic in `LogsdbIndexModeSettingsProvider`, seems like a better fit.

Reply:
Ok, I see. I missed that.